### PR TITLE
Add nanoflow event

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: src/components/BadgeButtonContainer.ts [hunk 1]
```diff
@@ -150,6 +154,15 @@ export default class BadgeButtonContainer extends Component<BadgeButtonContainer
                     guids: [ mxObject.getGuid() ]
                 }
             });
+        } else if (onClickEvent === "callNanoflow" && nanoflow) {
+            window.mx.data.callNanoflow({
+                context,
+                error: (error: Error) => {
```
Comment (djkwagala) [inline src/components/BadgeButtonContainer.ts]: Please follow the format on line 151

Comment (djkwagala) [reply src/components/BadgeButtonContainer.ts]: Also apply similar feedback like for `boolean-slider` and `badge`

**Review by djkwagala — CHANGES_REQUESTED** [review]

File: src/components/BadgeButtonContainer.ts (hunk 1 above)
Comment (jseremba) [reply src/components/BadgeButtonContainer.ts]: Done

File: src/BadgeButton.webmodeler.ts [hunk 2]
```diff
@@ -45,10 +45,16 @@ export function getPreviewCss() {
...
         visibilityMap.page = false;
     } else if (props.onClickEvent === "callMicroflow") {
         visibilityMap.page = false;
+        visibilityMap.nanoflow = false;
+    } else if (props.onClickEvent === "callNanoflow") {
+        visibilityMap.page = false;
+        visibilityMap.microflow = false;
     } else if (props.onClickEvent === "showPage") {
         visibilityMap.microflow = false;
+        visibilityMap.nanoflow = false;
```
Comment (Andries-Smit) [inline src/BadgeButton.webmodeler.ts]: Should be done the other way around
``` ts
visibilityMap.microflow = props.onClickEvent === "callMicroflow";
visibilityMap.nanoflow = props.onClickEvent === "callNanoflow";
visibilityMap.page = props.onClickEvent === "showPage"
```
lis code less if statements, future proof

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/BadgeButton.webmodeler.ts (hunk 2 above)
Comment (jseremba) [reply src/BadgeButton.webmodeler.ts]: Done

Comment (jseremba) [reply src/BadgeButton.webmodeler.ts]: Done

**Pull request merged**